src/compilers/v2/compiler_v2.cpp

```diff
@@ -0,0 +1,518 @@
+    virtual std::unique_ptr<ProgramBinary> Link(LinkerArgs const& args, Logger const& logger) const final;
+    virtual std::unique_ptr<ProgramBinary> Load(const void *data, size_t size) const final;
+    virtual std::unique_ptr<CompiledDxil> GetKernel(const char *name, ProgramBinary const& obj, CompiledDxil::Configuration const *, Logger const *logger) const final;
+    virtual std::byte * CopyWorkProperties(std::byte *WorkPropertiesBuffer, WorkProperties const& props) const final;
+    virtual size_t GetWorkPropertiesChunkSize() const final;
+    virtual uint64_t GetVersionForCache() const final;
+};
+
+class ProgramBinaryV2 : public ProgramBinary
+{
+public:
+    using unique_obj = unique_object<clc_binary>;
+
+private:
+    unique_obj m_Object;
+    unique_object<clc_parsed_spirv> m_Parsed;
+
+public:
+    ProgramBinaryV2(unique_obj obj);
+    clc_binary const& GetRaw() const { return m_Object; }
+    clc_parsed_spirv const& GetParsedInfo() const { return m_Parsed; }
+
+    // Inherited via ProgramBinary
+    virtual ~ProgramBinaryV2() = default;
+    virtual bool Parse(Logger const *logger) final;
+    virtual size_t GetBinarySize() const final;
+    virtual const void *GetBinary() const final;
+};
+
+class CompiledDxilV2 : public CompiledDxil
+{
+public:
+    using unique_obj = unique_object<clc_dxil_object>;
+
+private:
+    unique_obj m_Object;
+
+public:
+    CompiledDxilV2(ProgramBinaryV2 const& parent, unique_obj obj);
+    clc_dxil_object const& GetRaw() { return m_Object; }
+
+    // Inherited via CompiledDxil
+    virtual ~CompiledDxilV2() = default;
+    virtual size_t GetBinarySize() const final;
+    virtual const void *GetBinary() const final;
+    virtual void *GetBinary() final;
+};
+
+static clc_logger ConvertLogger(Logger const& logger)
+{
+    auto log = [](void *ctx, const char *msg) { static_cast<Logger*>(ctx)->Log(msg); };
+    clc_logger ret;
+    ret.error = log;
+    ret.warning = log;
+    ret.priv = (void*)&logger;
+    return ret;
+}
+
+#if 0
+#endif
+
+CompilerV2::CompilerV2(XPlatHelpers::unique_module compiler)
+    : m_Compiler(std::move(compiler))
+{
+#define GET_FUNC(Member, api) Member = m_Compiler.proc_address<decltype(&api)>(#api)
+    GET_FUNC(LoadLibclc, clc_libclc_new);
+    GET_FUNC(SerializeLibclc, clc_libclc_serialize);
+    GET_FUNC(DeserializeLibclc, clc_libclc_deserialize);
+    GET_FUNC(FreeSerializedLibclc, clc_libclc_free_serialized);
+    GET_FUNC(FreeLibclc, clc_free_libclc);
+    GET_FUNC(CompileImpl, clc_compile_c_to_spirv);
+    GET_FUNC(LinkImpl, clc_link_spirv);
+    GET_FUNC(FreeSpirv, clc_free_spirv);
+    GET_FUNC(ParseSpirv, clc_parse_spirv);
+    GET_FUNC(FreeParsedSpirv, clc_free_parsed_spirv);
+    GET_FUNC(GetKernelImpl, clc_spirv_to_dxil);
+    GET_FUNC(FreeDxil, clc_free_dxil_object);
+    GET_FUNC(GetCompilerVersion, clc_compiler_get_version);
+#undef GET_FUNC
+
+    if (!LoadLibclc ||

```

Comment:
Not all the GET_FUNC results are checked here?

Reply:
Good catch, the v2 compiler should implement everything.